src/views/GroupView.vue

```diff
@@ -1,73 +1,669 @@
+      }
+      for (const station of this.stations.items) {
+        if (!pendingStationsByName.hasOwnProperty(station.name)) {
+          diff[station.name] = {
+            ...station,
+            _rowVariant: "retire-item",
+            action: "retire",
+          };
+        } else {
+          const updatedStation = pendingStationsByName[station.name];
+          const EPSILON = 0.000000000001;
+          if (
+            Math.abs(updatedStation.latitude - station.latitude) > EPSILON ||
+            Math.abs(updatedStation.longitude - station.longitude) > EPSILON
+          ) {
+            let { latitude, longitude } = updatedStation;
+            if (Math.abs(latitude - station.latitude) > EPSILON) {
+              latitude = `<del>${station.latitude}</del> -> ${latitude}`;
+            }
+            if (Math.abs(longitude - station.longitude) > EPSILON) {
+              longitude = `<del>${station.longitude}</del> -> ${longitude}`;
+            }
+            diff[station.name] = {
+              ...updatedStation,
+              latitude,
+              longitude,
+              _rowVariant: "update-item",
+              action: "update",
+            };
+          } else {
+            diff[station.name] = {
+              ...updatedStation,
+              _rowVariant: "no-change-item",
+              action: "no change",
+            };
+          }
+        }
+      }
+      for (const station of this.stations.pendingStations) {
+        if (!diff.hasOwnProperty(station.name)) {
+          diff[station.name] = {
+            ...station,
+            _rowVariant: "add-item",
+            action: "add",
+          };
+        }
+      }
+      return (Object.values(diff) as StationData[]).sort((a, b) =>
+        a.name.localeCompare(b.name)
+      );
     },
   },
-  created: async function () {
-    await this.fetchGroup();
+  created() {
+    this.fetchGroup();
+    this.fetchStations();
   },
   methods: {
-    fetchGroup: async function () {
-      await this.$store.dispatch(
-        "Groups/GET_GROUP",
-        this.$route.params.groupname
-      );
+    // Stations:
+    dragCsvFileOver(event: DragEvent) {
+      this.stations.draggingCsvOver = true;
+      event.dataTransfer.dropEffect = "none";
+    },
+    dragCsvFileOut() {
+      this.stations.draggingCsvOver = false;
+    },
+    async droppedStationsCsvFile(event: DragEvent) {
+      this.stations.draggingCsvOver = false;
+      const csvText = await event.dataTransfer.files[0].text();
+      await this.parseStationsCsv(csvText);
+    },
+    async gotStationsCsvFile(event: Event) {
+      const file: File = (event.target as HTMLInputElement).files[0];
+      const csvText = await file.text();
+      await this.parseStationsCsv(csvText);
+    },
+    async parseStationsCsv(csvText: string) {
+      const monitoring = await csv().fromString(csvText);
+      this.stations.pendingStations = monitoring
+        .filter((item) => item.Type === "Camera")
```

Comment:
FYI There is also a monitoring type in TrapNZ called 'thermal camera' but we have ours listed as camera.